### PR TITLE
#212 Frontend - Select variables and their units to display in graph.

### LINKED ISSUE #212: [Frontend] - Select Variables and their Units to View in Graph Feature
As a User, I would like to be able to change which variables of a datasets I see on an axis as well as which units.

**Description** :
Within a dataset, there are multiple variables (Cell width, initial temperature, etc).  The goal of this feature is to allow the user to decide which of these variable will display on which axis. One of the issues is deciding how to deal with the unshared variables between the datasets. 
For example, if the user would like to display the cell width on the X axis but Dataset-2 does not have values for this variable, how would it be displayed?
The solution is to create a table similar to the one below: 
Following Datasets are omitted because they do not have “X” variable 
|-----------------------------------------------------------------------------------------------------|
|       |
| Dataset 1                                                                                                                       |
| Dataset 6 *                                                                                                                    |

Following Datasets are omitted because they do not have “Y” variable     
|   -------------------------------------------------------------------------------------------------   |
| Dataset 4                                                                                                                       |
| Dataset 6 *                                                                                                                    |
|*  A dataset may be missing both the X and Y variable|

The other issue is the selection of units for the variables. This requires methods to convert the units (if libraries do not exist) to the appropriate units. The units available for each variable should be predetermined (specific units for specific variables).

Pressure: kPa, MPa, GPa, bar, atm
Temperature: K (Kelvin), oC (celsius)
Velocity: km/s, mm/mu.s, m/s
Cell size, cell length (any physical lenght): mm, cm, m
density: kg/m^3, g/cc
specific volume: cc/g, m^3/kg

Conversion of the units will happen in the backend.

**Mockup** 
image
image

**Acceptance Criteria**
- Must display which datasets are not shown on the graph (which variable they lack)
- Must change the axis titles when a new one is selected
- Must show the appropriate units and convert 
- Must update the graph datapoints
- Must find a way to keep the dataset information in the frontend for these changes (Do not ask the database each time the user changes a variable)

### PR DESCRIPTION
**Pull request by lposo**
This is the frontend portion of this issue. Users can now swap between all available variables for each dataset loaded, and can view which datasets are missing the selected variable. They also have the option to change the units for each variable, though this feature is currently hardcoded in and does not actually convert units yet. More clarification is needed on how the stakeholder wants that to be handled in case of future units being added to the system.

Image of graph displaying initial pressure vs initial temperature:
image

Image of graph displaying initial pressure vs percent He:
image

An additional note is that due to ongoing changes to the graphing library used, the ability to see axis titles has not been completed on this branch. However, a different branch using the updated graph library has this feature implemented.

### CODE REVIEW

File: Client/src/Components/Graph/GraphView.tsx [hunk 1]
```diff
@@ -205,9 +321,94 @@ export default function GraphView() {
...
+                      <InputLabel id="xVariable">X Variable</InputLabel>
+                      <Select
+                        labelId="xVariable"
+                        id="xVariable"
+                        value={xVariableName}
+                        autoWidth={true}
+                        onChange={handleXVariableChange}
+                      >
+                        {variables.map(type => (
+                          <MenuItem value={type}>{type}</MenuItem>
```
Comment (ds161833) [inline Client/src/Components/Graph/GraphView.tsx]: can all this code be in its own component, we need to clean up current graphView its very heavy

File: Client/src/Models/Datasets/ITemporaryModel.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+export interface IDataPointModel {
```
Comment (ds161833) [inline Client/src/Models/Datasets/ITemporaryModel.ts]: what is this modal for, also the interface name is IDataPointModel but the file's name is ITemporaryModel

Comment (ds161833) [reply Client/src/Models/Datasets/ITemporaryModel.ts]: model*

Comment (lposo) [reply Client/src/Models/Datasets/ITemporaryModel.ts]: That was something I was using before, but it's not necessary now. I'll delete it.

Comment (TiraFire) [reply Client/src/Models/Datasets/ITemporaryModel.ts]: Reminder to remove it as well from the imports

File: Client/src/Components/Graph/GraphView.tsx [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
 import { DatasetsList } from "./DatasetsList"
 import Graph from './Graph'
 import { IDataPointExtremes } from "../../Models/Graph/IDataPointExtremes"
+import { IVariableUnits } from '../../Models/Datasets/IVariableModel'
+import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown'
+import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp'
 import SearchView from '../Search/SearchView'
-import { exampleExportDatasetModel } from '../../Models/Datasets/IDatasetModel'
 import { classStyles } from "../../appTheme"
+import { exampleDataset } from "../../Models/Datasets/ITemporaryModel"
```
Comment (TiraFire) [inline Client/src/Components/Graph/GraphView.tsx]: Reminder to remove this if the ITemporaryModel is removed

File: Client/src/Components/Graph/GraphView.tsx (hunk 1 above)
Comment (TiraFire) [reply Client/src/Components/Graph/GraphView.tsx]: Small thing, add border? Looks a bit strange to just have it pop out of nowhere. 
image

**Review by TiraFire — APPROVED** [review]
Good job, and yes with the refactoring that dima will be implementing, some of the acceptance criteria will be included later

File: Client/src/Components/Graph/AxesControl.tsx [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                                                <MenuItem value={type}>{type}</MenuItem>
+                                            ))}
+                                        </Select>
+                                    </FormControl>
+                                </Grid>
+                                <Grid item xs={4}>
+                                    <Typography align="center">
+                                        Datasets Missing X: {xVariableMissing.toString()}
+                                    </Typography>
+                                </Grid>
```
Comment (ds161833) [inline Client/src/Components/Graph/AxesControl.tsx:179]: peeeeeerfect, thi sis what im talking about, very nice ,very easy for me to integrate, good job

**Review by ds161833 — APPROVED** [review]

**Pull request merged**